Refactor:  Move the patient and encounter related API routes  to patientApi and encounterApi and type clean up

yash-learner opened the pull request:
## Proposed Changes

- Fixes first sub-task in #12909 


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Expanded patient and encounter APIs to support additional endpoints for searching, user management, questionnaire responses, observations, notes, and organization management.
  * Enhanced type definitions for patient and encounter data, introducing more detailed and structured types.

* **Refactor**
  * Updated all components and pages to use new `PatientRead` and `EncounterRead` types for improved data consistency and clarity.
  * Replaced generic API route usage with more specific, modular API imports throughout the codebase.

* **Bug Fixes**
  * Ensured API calls use defined string values for facility IDs, preventing potential issues with undefined parameters.

* **Chores**
  * Removed deprecated and redundant patient-related API routes and type imports to streamline the codebase.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yash-learner committed:
Move the patient related api routes  to patientApi from central routes

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Move encounter API routes to encounterAPI from central routes file

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Handle facility id being absent for some routes

yash-learner committed:
Remove accidently added vibe coding logfile of cursor

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

gigincg commented on src/types/emr/patient/patientApi.ts:
```diff
@@ -27,6 +42,19 @@ export default {
+    path: "/api/v1/patient/search/",
+    method: HttpMethod.POST,
+    TRes: Type<PatientSearchResponse>(),
+    TBody: Type<{
```
Make this an interface instead

gigincg commented on src/types/emr/encounter/encounterApi.ts:
```diff
@@ -1,14 +1,54 @@
+    path: "/api/v1/encounter/",
+    method: HttpMethod.POST,
+    TRes: Type<Encounter>(),
+    TBody: Type<EncounterEditRequest>(),
```
Shouldn't this be EncounterCreate

gigincg commented on src/types/emr/encounter/encounterApi.ts:
```diff
@@ -1,14 +1,54 @@
+    path: "/api/v1/encounter/{id}/",
+    method: HttpMethod.PUT,
+    TRes: Type<Encounter>(),
+    TBody: Type<EncounterEditRequest>(),
```
Let's keep a uniform naming strategy

EncounterEdit
EncounterCreate
EncounterRead 

should work.

cc: @bodhish

yash-learner replied:
@gigincg 

We will have EncounterBase as well ?

In new types, we are calling it Base -> `SupplyDeliveryBase`, `InvoiceBase`  and in old ones and at some places we are just calling entity name `Encounter`, `Patient`

Should we stick with one here as well, or is it fine to go with anyone ?

cc: @bodhish

yash-learner replied:
@gigincg 

Similar to other new types, I have added the common base(EncounterBase) type in encounter.ts and updated all relevant files and renamed `Encounter` to `EncounterRead`

https://github.com/ohcnetwork/care_fe/pull/12937/commits/16fb9a830db9ebd65945d2a071c686c85bc2d690

gigincg commented on src/types/emr/patient/patientApi.ts:
```diff
@@ -39,6 +67,97 @@ export default {
+      type: "header",
+    },
+  },
+  otpCreatePatient: {
```
OTP Based APIs ought to be separately managed, let's not keep them together with Patient APIs.

This has been split as separate issues as well

yash-learner committed:
Make the search and search_retrieve body's an interface

yash-learner committed:
Follow same naming pattern for types

Replace the depreceted intialFocus with autoFocus

yash-learner committed:
Revert changes related to OTP routes

yash-learner committed:
Remove dead code related to params in api types

I do not see any field related to the params in APIRoute definiton in src/Utils/request/types.ts

yash-learner committed:
Add base encounter, update reamining encounter types accordingly and update it relevant files

yash-learner committed:
Update remaining relevant files to use EncounterRead

yash-learner committed:
Update the patient type to have base and read and update relevant files and types

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Update it to use PatientRead

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

gigincg commented on src/components/Patient/PatientIndex.tsx:
```diff
@@ -168,8 +168,8 @@ export default function PatientIndex({ facilityId }: { facilityId: string }) {
-          year_of_birth: (patient as Patient).year_of_birth.toString(),
-          partial_id: (patient as Patient).id.slice(0, 5),
+          year_of_birth: (patient as PatientRead).year_of_birth.toString(),
+          partial_id: (patient as PatientRead).id.slice(0, 5),
```
Can we solve this? Not have a type assertion here?

yash-learner replied:
https://github.com/ohcnetwork/care_fe/pull/12937/commits/0c529fe070bb4e8f2ebb51456787a25f5d25adce

gigincg commented on src/components/Patient/allergy/list.tsx:
```diff
@@ -35,7 +35,7 @@ interface AllergyListProps {
   className?: string;
   readOnly?: boolean;
-  encounterStatus?: Encounter["status"];
+  encounterStatus?: EncounterRead["status"];
```
I would import and use EncounterStatus here

yash-learner replied:
https://github.com/ohcnetwork/care_fe/pull/12937/commits/06f8e871051f815c34f960cace894c56f31a4960

gigincg commented on src/types/emr/encounter/encounter.ts:
```diff
@@ -200,19 +200,26 @@ export type LocationHistory = {
+export interface EncounterRead
+  extends Omit<EncounterBase, "patient" | "facility"> {
+  patient: PatientRead;
+  facility: {
```
I don't think this is correct, let's clear this

yash-learner replied:
> I don't think this is correct, let's clear this

Which one, I didn't get what is incorrect, can you elaborate

If we are talking about `facility` and `patient` then it's correct backend returns the

```
curl 'https://care-api.do.ohc.network/api/v1/encounter/850c98e0-3aba-4b9c-bf3e-9591327370c0/?patient=68c26145-5ce7-4e1e-a59c-861923a1d2ba' \
```

**response**


<img width="1529" height="392" alt="image" src="https://github.com/user-attachments/assets/cf3f020d-7904-4ccb-8943-b5dc2bfd80b8" />

gigincg commented on src/types/emr/encounter/encounter.ts:
```diff
@@ -224,34 +231,17 @@ export interface Encounter {
-  external_identifier?: string;
-  facility: string;
-  discharge_summary_advice?: string | null;
+  discharge_summary_advice?: string;
```
This looks off. Can you check whether this field is relevant?

yash-learner replied:
https://github.com/ohcnetwork/care_fe/pull/12937/commits/e170fd69566977d8963b69044cfbd7e4586a6c38

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Remove type assertion

yash-learner committed:
Use EncounterStatus type

yash-learner committed:
Remove discharge summary advice from EncounterCreate

yash-learner committed:
Remove some of type assertions, typescript did not shout when removing them

yash-learner committed:
Fix more type assertions

yash-learner committed:
Remove organizations field from edit based on backend spec care/emr/resources/encounter/spec.py

and set hospitalization to undefined when it  is empty

yash-learner committed:
Backend is not updating hospitalization when not sending it

Make type Hospitalization type match BE type
https://github.com/ohcnetwork/care/blob/c7810d3ab5aa80683769036f2254736283752f36/care/emr/resources/encounter/spec.py#L41

yash-learner committed:
Transform read to edit and remove type assertion

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

yash-learner committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1

gigincg committed:
Merge branch 'develop' into issues/12909/migrate-from-cr-osr-pr1
